script/clone_job.pl

```diff
@@ -202,10 +213,16 @@ sub clone_job {
         $job->{settings}->{_START_AFTER_JOBS} = join(',', @new_chained)  if @new_chained;
     }
 
-    for my $type (keys %{$job->{assets}}) {
+  ASSET: for my $type (keys %{$job->{assets}}) {
         next if $type eq 'repo';    # we can't download repos
         for my $file (@{$job->{assets}->{$type}}) {
             my $dst = $file;
+            unless ($options{'skip-deps'} || $options{'skip-chained-deps'}) {
+                my @parents = map { get_job($_) } @{$job->{parents}->{Chained}};
```

Comment:
Is it possible to move this line before the first for loop (i.e. to L215)? The `$job` is not changing so IMO it's wasteful to recalculate `@parents` each iteration.